### DISCUSSION TITLE
Problematic API design when creating assets using Base64 URLs

### DISCUSSION THREAD

**Discussion by joshuabaker** [craftcms/cms#8882] [Ideas]
Unless I’m missing something, it’s currently not possible to related existing assets at the same time as creating new assets using Base64 URLs.

## Current implementation

**✅  Creating new asset using Base64 URL**

```json
{
    "fields": {
        "assetField": {
            "data": ["<Base64Url>"],
            "filename": ["filename.ext"]
        }
    }
}
```

**✅ Relating an existing asset**

```json
{
    "fields": {
        "assetField": [
            111,
            222,
            333
        ]
    }
}
```

**❌ Combination of the above**

You cannot be related and create in the same request.

## Proposed structure

Allow developers to create an asset within the array on the fly.

```json
{
    "fields": {
        "assetField": [
            111,
            {
                "data": "<Base64Url>",
                "filename": "filename.ext"
            },
            333
        ]
    }
}
```

**Comment by brandonkelly** [comment]
I don’t think you’re missing anything; we just never made this possible. Considering this a feature request.

**Comment by joshuabaker** [comment]
Thanks, @brandonkelly. Happy to help with the API design here, if needed.
